libraries/rush-lib/scripts/copyEmptyModules.js

```diff
@@ -22,37 +21,27 @@ module.exports = {
     // creates a problem when a `.d.ts` file references a module that doesn't have runtime code (i.e. -
     // a `.d.ts` file that only contains types).
     //
-    // This script looks through the `lib-esnext` folder for `.js` files that were produced by the TypeScript
+    // This script looks through the `lib-esm` folder for `.js` files that were produced by the TypeScript
     // compiler from `.ts` files that contain no runtime code and generates stub `.js` files for them in the
     // `lib` folder and copies the corresponding `.d.ts` files to the `lib`. This ensures that the `.d.ts`
     // files that end up in the `lib` folder don't have any unresolved imports. This is tested by the
     // `rush-lib-declaration-paths-test` project in the `build-tests`
 
     function stripCommentsFromJsFile(jsFileText) {
-      const lines = jsFileText.split('\n');
-      const resultLines = [];
-      for (const line of lines) {
-        const trimmedLine = line.trim();
-        if (trimmedLine === '' || trimmedLine.startsWith('//')) {
-          continue;
-        }
-
-        resultLines.push(trimmedLine);
-      }
-
-      return resultLines.join('\n');
+      jsFileText = jsFileText.replace(/^\s*\/\/.*$/gm, '');
+      jsFileText = jsFileText.replace(/\/\*.*\*\//gs, '');
```

Comment:
This regex will select:
```js
/* first */
some code
/* second */
```
and turn it into an empty string, because of the greedy matcher.